Life.java

```diff
@@ -1,55 +1,174 @@
 public class Life implements ILife {
+  private int[][] gameBoardCurrent;
+  private int[][] gameBoardNextGen;
```

Comment:
Sie haben hier einen recht komplizierten State Ihres Objekts mit zwei Arrays. "Moderner" wäre es, nicht alles in einem Life-Objekt abbilden zu wollen, sondern bei nextGeneration einfach ein neues Life-Objekt zu erzeugen und das alte zu verwerfen. Das würde Ihren State deutlich vereinfachen.